app/views/govuk_publishing_components/components/_layout_footer.html.erb

```diff
@@ -6,15 +6,37 @@
   with_border ||= false
   classes = %w(gem-c-layout-footer govuk-footer)
   classes << "gem-c-layout-footer--border" if with_border
+
+  ga4_index_total = 0
+  ga4_index_section_count = 0
+  amount_of_meta_sections = 0
+
+  navigation.each do |section|
+    ga4_index_section_count += 1
+    if section[:items]
+      ga4_index_total += section[:items].length
+    end
+  end
+
+  if meta.any?
+    amount_of_meta_sections = 1
+    ga4_index_section_count += 1
+    ga4_index_total += meta[:items].length
+  end
+
+  # For our GOVUK footer links, we need to add 2 to the index_total and index_section_count. This is due to the "Open Government License" and "Crown Copyright" links being defined separately to the rest of our footer links.
+  ga4_index_total += 2
+  ga4_index_section_count += 2
```

Comment:
There's quite a bit of code being introduced into the view here (also some repetition of code around lines 80 and 135). I think this would be better in a separate helper file.

Reply:
@andysellick Thanks, I've moved this to a helper file.